pysam/libcbcf.pyi

```diff
@@ -0,0 +1,457 @@
+    def __bool__(self) -> bool: ...
+    def __getitem__(self, index) -> VariantHeaderRecord: ...
+    def __iter__(self) -> Iterator[VariantHeaderRecord]: ...
+
+class VariantMetadata:
+    @property
+    def header(self) -> VariantHeader: ...
+    @property
+    def name(self) -> str: ...
+    # @property  # should this be exposed?
+    # def id(self) -> int: ...
+    @property
+    def number(self) -> Optional[str]: ...
+    @property
+    def type(self) -> Optional[str]: ...
+    @property
+    def description(self) -> Optional[str]: ...
+    @property
+    def record(self) -> Optional[VariantHeaderRecord]: ...
+    def remove_header(self) -> None: ...
+
+class VariantHeaderMetadata:
+    @property
+    def header(self) -> VariantHeader: ...
+    def add(
+        self,
+        id: str,
+        number: Optional[str],
+        type: Optional[str],
+        description: str,
+        **kwargs
+    ) -> None: ...
+    def __len__(self) -> int: ...
+    def __bool__(self) -> bool: ...
+    def __getitem__(self, key) -> VariantMetadata: ...
+    def remove_header(self, key: str) -> None: ...
+    def clear_header(self) -> None: ...
+    def __iter__(self) -> Iterator[str]: ...
+    def get(self, key: str, default: _D = ...) -> Union[_D, str]: ...
+    def __contains__(self, key) -> bool: ...
+    def iterkeys(self) -> Iterator[str]: ...
+    def itervalues(self) -> Iterator[Optional[str]]: ...
+    def iteritems(self) -> Iterator[Tuple[str, Optional[str]]]: ...
+    def items(self) -> List[Tuple[str, Optional[str]]]: ...
+    def keys(self) -> List[str]: ...
+    def values(self) -> List[Optional[str]]: ...
+
+class VariantContig:
+    @property
+    def header(self) -> VariantHeader: ...
+    @property
+    def name(self) -> str: ...
+    @property
+    def id(self) -> int: ...
+    @property
+    def length(self) -> Optional[int]: ...
+    @property
+    def header_record(self) -> VariantHeaderRecord: ...
+    def remove_header(self) -> None: ...
+
+class VariantHeaderContigs:
+    @property
+    def header(self) -> VariantHeader: ...
+    def __len__(self) -> int: ...
+    def __bool__(self) -> bool: ...
+    def __getitem__(self, key: str) -> VariantContig: ...
+    def remove_header(self, key: Union[str, int]) -> None: ...
+    def clear_header(self) -> None: ...
+    def __iter__(self) -> Iterator[str]: ...
+    def get(self, key: str, default: _D = ...) -> Union[_D, VariantContig]: ...
+    def __contains__(self, key: str) -> bool: ...
+    def iterkeys(self) -> Iterator[str]: ...
+    def itervalues(self) -> Iterator[Optional[VariantContig]]: ...
+    def iteritems(self) -> Iterator[Tuple[str, Optional[VariantContig]]]: ...
+    def keys(self) -> List[str]: ...
+    def items(self) -> List[Tuple[str, Optional[VariantContig]]]: ...
+    def values(self) -> List[Optional[VariantContig]]: ...
+    def add(self, id: str, length: Optional[int] = ..., **kwargs) -> None: ...
+
+class VariantHeaderSamples:
+    @property
```

Comment:
As above -- Sequence[str]?

Reply:
Same here, I left this as is because VariantHeaderSamples doesn’t implement all the methods that a Sequence requires.